game_data_cleaner/processing_functions.py

```diff
@@ -0,0 +1,188 @@
+import pandas as pd
+import numpy as np
+import requests
+import regex as re
+import time
+import os
+import gc
+import json
+
+# from statistics import mean
+
+# # NLP tools
+# import spacy
+
+# nlp = spacy.load("en_core_web_sm")
+# import re
+# import nltk
+# from sklearn.feature_extraction.text import CountVectorizer, TfidfVectorizer
+# from nltk.tokenize import word_tokenize
+
+
+def integer_reduce(data, columns, fill_value=0):
+    """
+    Reduces an integer type to its smallest memory size type
+
+    Inputs:
+    data: dataframe to reduce
+    columns: columns to reduce
+    fill_value: fill value to use if none
+
+    Returns:
+    data: dataframe with memory reduced data types
+    """
+    for column in columns:
+        print(data[column].unique())
+        data[column] = data[column].astype("Int64")
+        print(column)
+        data[column].fillna(fill_value, inplace=True)
+        if (data[column].max() <= 127) & (data[column].min() >= -128):
+            data[column] = data[column].astype("int8")
+        elif (data[column].max() <= 32767) & (data[column].min() >= -32768):
+            data[column] = data[column].astype("int16")
+        elif (data[column].max() <= 2147483647) & (data[column].min() >= -2147483648):
+            data[column] = data[column].astype("int32")
+
+    return data
+
+
+# def text_block_processor(text):
```

Comment:
Is the below stuff that will no longer be used or something you plan to implement in the future?  I'm generally opposed to large commented code blocks in main (yeah, I don't always follow this well), but totally makes sense if this is a WIP